src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferAsSpanOverSubstring.cs

```diff
@@ -0,0 +1,312 @@
+                    return -1;
+
+                int replacementCount = 0;
+                foreach (var parameter in candidate.Parameters)
+                {
+                    if (isSubstringLookup[parameter.Ordinal] && SymbolEqualityComparer.Default.Equals(parameter.Type, symbols.RoscharType))
+                    {
+                        ++replacementCount;
+                        continue;
+                    }
+
+                    var oldParameter = method.Parameters[parameter.Ordinal];
+                    if (!SymbolEqualityComparer.Default.Equals(parameter.Type, oldParameter.Type))
+                    {
+                        return -1;
+                    }
+                }
+
+                return replacementCount;
+            }
+        }
+
+        private static IEnumerable<IMethodSymbol> GetAllAccessibleOverloadsIncludingSelf(IInvocationOperation invocation, CancellationToken cancellationToken)
+        {
+            var method = invocation.TargetMethod;
+            var model = invocation.SemanticModel;
+            int location = invocation.Syntax.SpanStart;
+            var instance = invocation.Instance;
+
+            IEnumerable<IMethodSymbol> allOverloads;
+            if (method.IsStatic)
+            {
+                allOverloads = model.LookupStaticMembers(location, method.ContainingType, method.Name).OfType<IMethodSymbol>();
+            }
+            else if (instance is not null)
+            {
+                var enclosingType = GetEnclosingType(model, location, cancellationToken);
+                allOverloads = model.LookupSymbols(location, instance.Type, method.Name).OfType<IMethodSymbol>();
+                if (DerivesFromOrEqualTo(instance.Type, enclosingType) || instance is IInstanceReferenceOperation)
+                {
+                    allOverloads = allOverloads.Union(model.LookupBaseMembers(location, method.Name).OfType<IMethodSymbol>());
+                }
+            }
+            //  This can happen when compiling invalid code.
+            else
+            {
+                allOverloads = Enumerable.Empty<IMethodSymbol>();
+            }
+
+            return allOverloads.Where(x => x.IsStatic == method.IsStatic && SymbolEqualityComparer.Default.Equals(x.ReturnType, method.ReturnType));
+
+            static INamedTypeSymbol GetEnclosingType(SemanticModel model, int location, CancellationToken cancellationToken)
+            {
+                ISymbol symbol = model.GetEnclosingSymbol(location, cancellationToken);
+                if (symbol is not INamedTypeSymbol type)
+                    type = symbol.ContainingType;
+
+                return type;
+            }
+        }
+
+        private static bool DerivesFromOrEqualTo(ITypeSymbol derived, ITypeSymbol candidateBase)
+        {
+            return derived.DerivesFrom(candidateBase, baseTypesOnly: true) || SymbolEqualityComparer.Default.Equals(derived, candidateBase);
```

Comment:
Looking at the implementation of `DerivesFrom` it seems to already handle the equals case.

Reply:
Fixed.